### PR TITLE
pass unit tests

### PR DESCRIPTION
**Pull request by hanfei1991**
fix tests

### CODE REVIEW

File: executor/builder.go [hunk 1]
```diff
@@ -2314,7 +2314,6 @@ func (b *executorBuilder) buildTableReader(v *plannercore.PhysicalTableReader) *
 	}
 
 	ts := v.GetTableScan()
-	logutil.BgLogger().Info("build table reader", zap.Int64("my table id", ts.Table.ID), zap.String("table name", ts.Table.Name.L), zap.String("range str", ts.Ranges[0].String()))
```
Comment (windtalker) [inline executor/builder.go:2317]: why delete this line?

File: planner/core/task.go [hunk 2]
```diff
@@ -525,12 +522,10 @@ func (p *PhysicalHashJoin) attach2Task(tasks ...task) task {
 	lTask := finishCopTask(p.ctx, tasks[0].copy())
 	rTask := finishCopTask(p.ctx, tasks[1].copy())
 	p.SetChildren(lTask.plan(), rTask.plan())
-	p.schema = BuildPhysicalJoinSchema(p.JoinType, p)
```
Comment (windtalker) [inline planner/core/task.go:528]: Why remove this line?

File: planner/core/task.go [hunk 3]
```diff
@@ -525,12 +522,10 @@ func (p *PhysicalHashJoin) attach2Task(tasks ...task) task {
...
 	task := &rootTask{
 		p:   p,
 		cst: lTask.cost() + rTask.cost() + p.GetCost(lTask.count(), rTask.count()),
 	}
-	logutil.BgLogger().Info("hash join cost", zap.Float64("hj cost", task.cst))
```
Comment (windtalker) [inline planner/core/task.go:533]: Ditto

File: planner/core/find_best_task.go [hunk 4]
```diff
@@ -1112,12 +1112,6 @@ func (ds *DataSource) convertToTableScan(prop *property.PhysicalProperty, candid
 		tblColHists:       ds.TblColHists,
 		cst:               cost,
 	}
-	if ts.StoreType == kv.TiFlash {
-		if prop.TaskTp != property.CopTiFlashLocalReadTaskType && prop.TaskTp != property.RootTaskType && prop.TaskTp != property.CopTiFlashGlobalReadTaskType {
-			logutil.BgLogger().Error("invalid task:", zap.String("type", prop.TaskTp.String()))
-			return invalidTask, nil
-		}
-	}
```
Comment (windtalker) [inline planner/core/find_best_task.go:1120]: So it is guaranteed in somewhere else that `CopTiFlashLocalReadTaskType` and `CopTiFlashGlobalReadTaskType` will not be generated with non-TiFlash storeType based datasource?

Comment (hanfei1991) [reply planner/core/find_best_task.go:1120]: Sometimes CopSingleReadTaskType can generate TiFlash Plan, we should not cut this search branch.

File: executor/builder.go (hunk 1 above)
Comment (hanfei1991) [reply executor/builder.go:2317]: Some test forbid to output any log during `insert` and `update`, see https://github.com/pingcap/tidb/blob/82829e1fa5b908cf4b6ea8c97ba14a38fcde81fa/executor/seqtest/seq_executor_test.go#L1436-L1435

File: planner/core/task.go (hunk 2 above)
Comment (hanfei1991) [reply planner/core/task.go:528]: Hash Join should not call `BuildPhysicalJoinSchema`, this line is added by this pr, but it's unnesessary. Hash join's schema should keep the same no matter which one is small table.

File: planner/core/find_best_task.go (hunk 4 above)
Comment (windtalker) [reply planner/core/find_best_task.go:1120]: OK， but better to return `invalidTask` if storeType is not TiFlash and task type is CopTiFlashxxx

Comment (hanfei1991) [reply planner/core/find_best_task.go:1120]: added in https://github.com/hanfei1991/tidb/blob/ed3065d287773a654e38e3faa4271417253dbd2e/planner/core/find_best_task.go#L380-L379

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**